tools/admin/src/main/scala/whisk/core/database/UserCommand.scala

```diff
@@ -164,14 +172,23 @@ class UserCommand extends Subcommand("user") with WhiskCommand {
     authStore
       .get[ExtendedAuth](DocInfo(create.subject()))
       .flatMap { auth =>
+        val nsToUpdate = create.desiredNamespace(authKey).name
+        var newNS = auth.namespaces.filter(_.namespace.name != nsToUpdate)
         if (auth.isBlocked) {
           Future.successful(Left(IllegalState(CommandMessages.subjectBlocked)))
-        } else if (auth.namespaces.exists(_.namespace.name == create.desiredNamespace(authKey).name)) {
-          Future.successful(Left(IllegalState(CommandMessages.namespaceExists)))
-        } else {
-          val newNS = auth.namespaces + WhiskNamespace(create.desiredNamespace(authKey), authKey)
+        } else if (!auth.namespaces.exists(_.namespace.name == nsToUpdate) || create.force.isSupplied) {
+          newNS = newNS.+(WhiskNamespace(create.desiredNamespace(authKey), authKey))
           val newAuth = WhiskAuth(auth.subject, newNS).revision[WhiskAuth](auth.rev)
           authStore.put(newAuth).map(_ => Right(authKey.compact))
+        } else if (create.revoke.isSupplied) {
+          val updatedAuthKey = auth.namespaces.find(_.namespace.name == nsToUpdate).get.authkey
```

Comment:
What if namespace to revoke does not exist? 

Okie it would then just add a new namespace entry with new AuthKey i.e. enter the previous flow in if. Should we return an error if namespace to revoke does not exist?

Reply:
I don't have a strong opinion on this, but the former sounds a bit simpler and consistent with `-f` behavior. wdyt?